### PR TITLE
Eosio build: 7 gigs of ram requirement and other fixes

### PR DESCRIPTION
**Pull request by pacificcode**
Set min ram requirements to 7 gigs (my dev servers only have 7 gigs).
Set boost version check for centos.
Set network response checks for curl and git.
Other minor changes.

### CODE REVIEW

File: tests/testUtils.py [hunk 1]
```diff
@@ -1769,8 +1769,7 @@ def discoverLocalNodes(self, totalNodes, timeout=0):
         nodes=[]
 
         pgrepOpts="-fl"
-        if platform.linux_distribution()[0] == "Ubuntu" or platform.linux_distribution()[0] == "LinuxMint" or platform.linux_distribution()[0] == "Fedora" or\
-                platform.linux_distribution()[0] == "arch":
+        if platform.linux_distribution()[0] in ["Ubuntu", "LinuxMint", "Fedora","CentOS Linux"]:
```
Comment (heifner) [inline tests/testUtils.py]: Can you add "arch" back in.

File: scripts/eosio_build_centos.sh [hunk 2]
```diff
@@ -176,18 +176,23 @@
...
+		ln -s ${HOME}/opt/cmake-3.10.2/ cmake
 		cd cmake
 		./bootstrap
 		if [ $? -ne 0 ]; then
 			printf "\tError running bootstrap for CMAKE.\n"
 			printf "\tExiting now.\n\n"
 			exit;
 		fi
-		make
+		make -j${CPU_CORE}
```
Comment (heifner) [inline scripts/eosio_build_centos.sh]: min((RAMinGB/2GB), CPU_CORE) would be better but not sure how easy that is to calculate.

File: scripts/eosio_build_amazon.sh [hunk 3]
```diff
@@ -16,8 +16,8 @@
 	printf "\tDisk space total: ${DISK_TOTAL}G\n"
 	printf "\tDisk space available: ${DISK_AVAIL}G\n"
 
-	if [ $MEM_MEG -lt 4000 ]; then
-		printf "\tYour system must have 4 or more Gigabytes of physical memory installed.\n"
+	if [ $MEM_MEG -lt 7000 ]; then
```
Comment (gleehokie) [inline scripts/eosio_build_amazon.sh]: Didn't we settle on 8 GB of RAM during the scrum today?

File: scripts/eosio_build_centos.sh [hunk 4]
```diff
@@ -21,21 +21,21 @@
 	printf "\tDisk space total: ${DISK_TOTAL%.*}G\n"
 	printf "\tDisk space available: ${DISK_AVAIL%.*}G\n"
 
-	if [ $MEM_MEG -lt 4000 ]; then
-		echo "Your system must have 4 or more Gigabytes of physical memory installed."
+	if [ $MEM_MEG -lt 7000 ]; then
```
Comment (gleehokie) [inline scripts/eosio_build_centos.sh]: 8 GB?

File: scripts/eosio_build_darwin.sh [hunk 5]
```diff
@@ -23,8 +23,8 @@
 	printf "\tDisk space total: ${DISK_TOTAL}G\n"
 	printf "\tDisk space available: ${DISK_AVAIL}G\n\n"
 
-	if [ $MEM_GIG -lt 8 ]; then
-		echo "Your system must have 8 or more Gigabytes of physical memory installed."
+	if [ $MEM_GIG -lt 7 ]; then
```
Comment (gleehokie) [inline scripts/eosio_build_darwin.sh]: 8 GB?

File: scripts/eosio_build_fedora.sh [hunk 6]
```diff
@@ -20,8 +20,8 @@
 	printf "\tDisk space total: ${DISK_TOTAL%.*}G\n"
 	printf "\tDisk space available: ${DISK_AVAIL%.*}G\n"
 
-	if [ $MEM_MEG -lt 4000 ]; then
-		printf "\tYour system must have 8 or more Gigabytes of physical memory installed.\n"
+	if [ $MEM_MEG -lt 7000 ]; then
```
Comment (gleehokie) [inline scripts/eosio_build_fedora.sh]: 8 GB?

File: scripts/eosio_build_ubuntu.sh [hunk 7]
```diff
@@ -22,8 +22,8 @@
 	printf "\tDisk space total: ${DISK_TOTAL%.*}G\n"
 	printf "\tDisk space available: ${DISK_AVAIL%.*}G\n"
 
-	if [ $MEM_MEG -lt 4000 ]; then
-		printf "\tYour system must have 8 or more Gigabytes of physical memory installed.\n"
+	if [ $MEM_MEG -lt 7000 ]; then
```
Comment (gleehokie) [inline scripts/eosio_build_ubuntu.sh]: 8 GB?

File: tests/testUtils.py (hunk 1 above)
Comment (pacificcode) [reply tests/testUtils.py]: Sorry bout that, arch is back in.

File: scripts/eosio_build_centos.sh (hunk 2 above)
Comment (pacificcode) [reply scripts/eosio_build_centos.sh]: Do you mean something like:
	MEM_GIG=$(( (($MEM_MEG / 1000) / 2) ))
	JOBS=$(( ${MEM_GIG} > ${CPU_CORE} ? ${CPU_CORE} : ${MEM_GIG} ))

File: scripts/eosio_build_amazon.sh (hunk 3 above)
Comment (pacificcode) [reply scripts/eosio_build_amazon.sh]: @gleehokie We explicitly and unquestionably decided in the presence of all that the minimum requirements for physical RAM would be 8 Gigs.
However ... my dev servers only have been 7.8 gigs of ram installed!
[bhamilton@ip-10-160-13-251 eos]$ free -m
              total        used        free      shared  buff/cache   available
Mem:           7821         193        5838          40        1788        7170
Swap:             0           0           0

File: scripts/eosio_build_centos.sh (hunk 2 above)
Comment (heifner) [reply scripts/eosio_build_centos.sh]: Sure if that works everywhere.

Comment (pacificcode) [reply scripts/eosio_build_centos.sh]: Implemented on CentOS and running a full dependency build now. Ill push it when I have a successful build.

**Comment by heifner** [comment]
7 GB ok'ed in scrum on 4/13/2018.

File: scripts/eosio_build_amazon.sh (hunk 3 above)
Comment (kenorb) [reply scripts/eosio_build_amazon.sh]: Can you please [update Docker Requirement for 7GB](https://github.com/EOSIO/eos/blob/master/Docker/README.md#docker-requirement)?

**Comment by arvoelke** [comment]
Note [the wiki documentation](https://github.com/EOSIO/eos/wiki/Local-Environment) still says `8 GB` despite the above change to `7 GB`. Also, at least with Ubuntu, if you only have `2` cores, then `4 GB` is plenty. Both comments have been documented in #2992.

**Pull request merged**